torch/testing/_internal/common_methods_invocations.py

```diff
@@ -599,22 +599,33 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad, **kw
 
     return inputs
 
+def sample_inputs_addmm_non_fusible_nodes(op_info, device, dtype, requires_grad, **kwargs):
+    tests_list = [
+        ((2, 3), (2, 2), (2, 3), kwargs.get('alpha', 1), kwargs.get('beta', 1), False)
+    ]
+    tests_with_lhs_broadcasting = [
+        ((1,), (2, 2), (2, 3), kwargs.get('alpha', 1), kwargs.get('beta', 1), True),
+        ((), (2, 2), (2, 3), kwargs.get('alpha', 1), kwargs.get('beta', 1), True)
+    ]
+    test_cases = tests_list + tests_with_lhs_broadcasting  # type: ignore
+    inputs = tuple(SampleInput(make_tensor(shape_a, device, dtype, requires_grad=requires_grad),
+                               args=(make_tensor(shape_b, device, dtype,
+                                                 requires_grad=requires_grad),
+                                     make_tensor(shape_c, device, dtype,
+                                                 requires_grad=requires_grad)),
+                               kwargs={'alpha': alpha, 'beta': beta},
+                               broadcasts_input=broadcasts_input)
+                   for shape_a, shape_b, shape_c, alpha, beta, broadcasts_input in test_cases)
+    return inputs
+
 def sample_inputs_addmm(op_info, device, dtype, requires_grad, **kwargs):
-    input = SampleInput(
-        make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
-        args=(
-            make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
-            make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=False)))
     if dtype.is_complex:
-        another_input = SampleInput(
-            make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
-            args=(
-                make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
-                make_tensor((S, S), device, dtype, low=None, high=None, requires_grad=False)),
-            kwargs=dict(beta=1 + 2j, alpha=2 + 3j))
-        return (input, another_input)
+        inputs = sample_inputs_addmm_non_fusible_nodes(op_info, device, dtype, requires_grad,
```

Comment:
See comment below for an alternative name suggestion. I would just create `sample_inputs_addmm` and if alpha and beta are unspecified I would populate the values in this function, and if they're specified I would use them. This should let this pr use the partial approach described below.